Panic if the error implements causer and the cause is nil

**Sample code**:
```
import (
	"fmt"
	"github.com/getsentry/raven-go"
	pkgErrors "github.com/pkg/errors"
)

type causer interface {
	Cause() error
}

type customErr struct {
	msg   string
	cause error
}

func (e *customErr) Error() (errorMsg string) {
	if e.msg != "" && e.cause != nil {
		errorMsg = fmt.Sprintf("%v \n\t==>> %v", e.msg, e.cause)
	} else if e.msg == "" && e.cause != nil {
		errorMsg = fmt.Sprintf("%v", e.cause)
	} else if e.msg != "" && e.cause == nil {
		errorMsg = fmt.Sprintf("%s", e.msg)
	}
	return
}

// Implementing the causer interface from github.com/pkg/errors
func (e *customErr) Cause() error {
	return e.cause
}

func main() {
	_err := &customErr{
		cause: nil,
		msg:   "This is a test",
	}
	err := pkgErrors.WithStack(_err)
	raven.CaptureErrorAndWait(err, nil)
}
```

**Output**:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x12eeec6]

goroutine 1 [running]:
github.com/getsentry/raven-go.NewException(0x0, 0x0, 0xc000323200, 0x3)
	/Users/adityasatyavada/go/src/github.com/getsentry/raven-go/exception.go:11 +0x26
github.com/getsentry/raven-go.(*Client).CaptureErrorAndWait(0xc000328000, 0x1440900, 0xc000323100, 0x0, 0x0, 0x0, 0x0, 0xc000323100, 0xc0003230e0)
	/Users/adityasatyavada/go/src/github.com/getsentry/raven-go/client.go:740 +0x2fd
github.com/getsentry/raven-go.CaptureErrorAndWait(0x1440900, 0xc000323100, 0x0, 0x0, 0x0, 0x0, 0xc000147f88, 0xc000074058)
	/Users/adityasatyavada/go/src/github.com/getsentry/raven-go/client.go:751 +0x70
main.main()
```

**Solution:**

[Current code](https://github.com/getsentry/raven-go/blob/master/client.go#L756):
`cause := Cause(err)`

**Proposed change:**

```
cause := pkgErrors.Cause(err)
if cause == nil{
	cause = err
}
```

--- a/client.go
+++ b/client.go
@@ -1074,8 +1074,15 @@ func init() {
 //     }
 //
 // If the error does not implement Cause, the original error will
-// be returned. If the error is nil, nil will be returned without further
+// be returned.
+//
+// If the cause of the error is nil, then the original
+// error will be returned.
+//
+// If the error is nil, nil will be returned without further
 // investigation.
+//
+// Will return the deepest cause which is not nil.
 func Cause(err error) error {
 	type causer interface {
 		Cause() error
@@ -1086,7 +1093,13 @@ func Cause(err error) error {
 		if !ok {
 			break
 		}
-		err = cause.Cause()
+
+		if _cause := cause.Cause(); _cause != nil {
+			err = _cause
+		} else {
+			break
+		}
+
 	}
 	return err
 }

--- a/client_test.go
+++ b/client_test.go
@@ -3,6 +3,7 @@ package raven
 import (
 	"encoding/json"
 	"fmt"
+	pkgErrors "github.com/pkg/errors"
 	"reflect"
 	"testing"
 	"time"
@@ -283,6 +284,41 @@ func TestCaptureNilError(t *testing.T) {
 	eventID := client.CaptureError(nil, nil)
 	if eventID != "" {
 		t.Error("expected empty eventID:", eventID)
+	}
+}
+
+// Custom error which implements causer
+type customErr struct {
+	msg   string
+	cause error
+}
+
+func (e *customErr) Error() (errorMsg string) {
+	if e.msg != "" && e.cause != nil {
+		errorMsg = fmt.Sprintf("%v \n\t==>> %v", e.msg, e.cause)
+	} else if e.msg == "" && e.cause != nil {
+		errorMsg = fmt.Sprintf("%v", e.cause)
+	} else if e.msg != "" && e.cause == nil {
+		errorMsg = fmt.Sprintf("%s", e.msg)
+	}
+	return
+}
+
+// Implementing the causer interface from github.com/pkg/errors
+func (e *customErr) Cause() error {
+	return e.cause
+}
+
+func TestCaptureNilCauseError(t *testing.T) {
+	var client = DefaultClient
+	err := pkgErrors.WithStack(&customErr{
+		// Setting a nil cause
+		cause: nil,
+		msg:   "This is a test",
+	})
+	eventID := client.CaptureError(err, nil)
+	if eventID == "" {
+		t.Error("expected non-empty eventID:", eventID)
 	}
 }
 

